File: metadata_preprocessing.py
import pandas as pd

BD_metadata = pd.read_excel("burgos_dbgap/burgos_dbgap_metadata.xlsx")
BD_metadata.rename(columns = {"submitted_subject_id": "subject_id",
                    "biospecimen_repository_sample_id": "sample_id",
                    "gender.M.1": "sex", "expired_age": "age",
                    "ApoE": "apoe", "AD": "group"}, inplace = True)
BD_metadata["apoe"] = BD_metadata["apoe"].str.replace("/", "")
BD_metadata["sex"] = BD_metadata["sex"].map({1: "male", 2: "female"})
BD_metadata["group"] = BD_metadata["group"].map({"yes": "AD", "no": "control"})
BD_metadata.loc[BD_metadata["sample_id"].str.contains("CSF"), "type"] = "CSF"
BD_metadata.loc[BD_metadata["sample_id"].str.contains("SER"), "type"] = "serum"
BD_metadata["study"] = "burgos_dbgap"


SS_sample_metadata = pd.read_excel("silver_seq/silver_seq_metadata.xlsx")
SS_sample_metadata.rename(columns = {"donor_id_alias": "subject_id",
                            "sample_id_alias": "sample_id",
                            "donor_group": "group"}, inplace = True)
SS_patient_metadata = pd.read_excel("silver_seq/silver_seq_patient_metadata.xlsx")
SS_patient_metadata.rename(columns = {"donor_id_alias": "subject_id", 
                            "expired_age": "age"}, inplace = True)
SS_metadata = pd.merge(SS_sample_metadata, SS_patient_metadata, on = "subject_id")
SS_metadata["sex"] = SS_metadata["sex"].map({"M": "male", "F": "female"})
SS_metadata["group"] = SS_metadata["group"].map({"AD": "AD", "N": "control"})
SS_metadata["type"] = "plasma"
SS_metadata["study"] = "silver_seq"

TD_metadata = pd.read_excel("toden/toden_metadata.xlsx")
TD_metadata.rename(columns = {"Run": "sample_id", "PatientID": "subject_id",
                    "Age": "age", "Gender": "sex", "Disease": "group", 
                    "Apoe.status": "apoe"}, inplace = True)
TD_metadata["apoe"] = TD_metadata["apoe"].str.replace("/", "").str.replace("E", "")
TD_metadata["group"] = TD_metadata["group"].map({"AD": "AD", "NCI": "control"})
TD_metadata["sex"] = TD_metadata["sex"].str.lower()
TD_metadata["type"] = "plasma"
TD_metadata["study"] = "toden"

columns = ["study", "sample_id", "subject_id", "group", "type", 
           "sex", "age", "apoe", "apoe_carrier", "apoe_dose"]
dfs = [BD_metadata[columns], SS_metadata[columns], TD_metadata[columns]]
metadata = pd.concat(dfs, axis = 0)
metadata.to_csv("combined_metadata.tsv", sep = "\t")

File: metadata_preprocessing.R
library(readxl)
library(dplyr)
library(stringr)
library(readr)

# BURGOS DBGAP
BD_metadata <- read_excel("burgos_dbgap/burgos_dbgap_metadata.xlsx") %>%
    rename(
        subject_id = submitted_subject_id,
        sample_id = biospecimen_repository_sample_id,
        sex = `gender.M.1`,
        age = expired_age,
        apoe = ApoE,
        group = AD
    ) %>% mutate(
        apoe = str_replace_all(apoe, "/", ""),
        sex = case_when(
            sex == 1 ~ "male", 
            sex == 2 ~ "female"
        ), group = case_when(
            group == "yes" ~ "AD", 
            group == "no" ~ "control"
        ), type = case_when(
            str_detect(sample_id, "CSF") ~ "CSF",
            str_detect(sample_id, "SER") ~ "serum"
        ), study = "burgos_dbgap"
    )

# SILVER SEQ
SS_sample_metadata <- read_excel("silver_seq/silver_seq_metadata.xlsx") %>%
    rename(
        subject_id = donor_id_alias,
        sample_id = sample_id_alias,
        group = donor_group
    )
 
SS_patient_metadata <- read_excel("silver_seq/silver_seq_patient_metadata.xlsx") %>%
    rename(
        subject_id = donor_id_alias,
        age = expired_age
    )
 
SS_metadata <- inner_join(SS_sample_metadata, SS_patient_metadata, by = "subject_id") %>%
    mutate(
        apoe = as.character(apoe),
        sex = case_when(
            sex == "M" ~ "male", 
            sex == "F" ~ "female"
        ), group = case_when(
            group == "AD" ~ "AD", 
            group == "N" ~ "control"
        ), type = "plasma",
        study = "silver_seq"
    )

# TODEN
TD_metadata <- read_excel("toden/toden_metadata.xlsx", na = c("", "None")) %>%
    rename(
        sample_id = Run,
        subject_id = PatientID,
        age = Age,
        sex = Gender,
        group = Disease,
        apoe = `Apoe.status`
    ) %>% mutate(
        subject_id = as.character(subject_id),
        apoe = apoe %>% 
            str_replace_all("/", "") %>% 
            str_replace_all("E", ""),
        group = case_when(
            group == "AD" ~ "AD", 
            group == "NCI" ~ "control"
        ), sex = str_to_lower(sex),
        type = "plasma",
        study = "toden"
    )

# COMBINED
columns <- c("study", "sample_id", "subject_id", "group", "type",
             "sex", "age", "apoe", "apoe_carrier", "apoe_dose")
 
metadata <- bind_rows(
  BD_metadata %>% select(all_of(columns)),
  SS_metadata %>% select(all_of(columns)),
  TD_metadata %>% select(all_of(columns))
)
 
write_tsv(metadata, "combined_metadata.tsv", na = "")